runtime/src/main/java/org/corfudb/runtime/clients/LogUnitClient.java

```diff
@@ -316,16 +318,31 @@ private static Object handleTrimMarkResponse(CorfuPayloadMsg<Long> msg,
     }
 
     /**
-     * Read data from the log unit server for a range of offsets of a particular stream.
+     * Read data from the log unit server for a range of addresses.
      *
-     * @param stream      Stream ID to be read.
-     * @param offsetRange Range of global offsets.
+     * @param range Range of global offsets.
      * @return CompletableFuture which returns a ReadResponse on completion.
      */
-    public CompletableFuture<ReadResponse> read(UUID stream, Range<Long> offsetRange) {
+    public CompletableFuture<ReadResponse> read(Range<Long> range) {
         Timer.Context context = getTimerContext("readRange");
         CompletableFuture<ReadResponse> cf = router.sendMessageAndGetCompletable(
-                CorfuMsgType.READ_REQUEST.payloadMsg(new ReadRequest(offsetRange, stream)));
+                CorfuMsgType.READ_REQUEST.payloadMsg(new ReadRequest(range)));
+        return cf.thenApply(x -> {
+            context.stop();
+            return x;
+        });
+    }
+
+    /**
+     * Read data from the log unit server for a list of addresses.
+     *
+     * @param list list of global addresses.
+     * @return CompletableFuture which returns a ReadResponse on completion.
+     */
+    public CompletableFuture<ReadResponse> read(List<Long> list) {
```

Comment:
ok I see the list is for the implicit order in which the responses are sent back.